testkit/javadsl/src/main/scala/com/lightbend/lagom/internal/testkit/ProducerStubFactory.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.testkit
```

Comment:
> When working in the javadsl I preferred moving anything suposed to be internal to the internal package.

That makes sense, but in other projects we sometimes combine both name spaces. For example, in persistence there are packages called:

- `com.lightbend.lagom.internal.persistence`
- `com.lightbend.lagom.javadsl.persistence`
- `com.lightbend.lagom.internal.javadsl.persistence`
- `com.lightbend.lagom.scaladsl.persistence`
- `com.lightbend.lagom.internal.scaladsl.persistence`

We could continue that pattern with `com.lightbend.lagom.internal.javadsl.testkit` for internal code that is specific to the Java DSL, reserving `com.lightbend.internal.testkit` for code that is shared between Java and Scala.

> Code looks very similar but almost all internal classes are stubs implementing public APIs (Subscriber and Topic) which differ from javadsl to scaladsl already.

That's what I meant about having thin bindings to language-specific APIs. I think it would be possible to share the implementation of the methods in generic traits but to put the `extends Subscriber[T]` etc. clauses into the language-specific subtypes. Does that make sense?